Cython/Compiler/PyrexTypes.py

```diff
@@ -1719,6 +1723,15 @@ def py_type_name(self):
         return "__import__(%r, None, None, ['']).%s" % (self.module_name,
                                                         self.name)
 
+    def __eq__(self, other):
```

Comment:
This fix is needed to fix failing test of `test_dataclasses`. The issue was strange. Cython complained about redeclared: `AssertionError: None != "272:15: 'list[User_TestCase_test_helper_asdict_builtin_containers]' redeclared"`

The issue surprisingly occurs **only** when both cpp and C tests are run at once (running the test is successful when `--no-c` or `--no-cpp` arguments are provided)

It seems that between the runs two different instances of `PyExtensionType` are created. This breaks following line: https://github.com/cython/cython/pull/7288/files#diff-7f63ffd84e67d3d113d7ad10fa110111c779a8ca5edd7e12dc51097b36405eabR4908

Reply:
Now fixed - see https://github.com/cython/cython/pull/7288/files#r2554302840